### DISCUSSION TITLE
selecting BlueALSA devices by name instead of address

### DISCUSSION THREAD

**Discussion by borine** [Arkq/bluez-alsa#656] [Ideas]
I've been using a minor change to the ALSA config file `20-bluealsa.conf.in` that allows me to specify devices by name in place of address in the "DEV" argument on the command line given simple entries in my `~/.asoundrc` file.

```diff
diff --git a/src/asound/20-bluealsa.conf.in b/src/asound/20-bluealsa.conf.in
index 3532c5c..08d010d 100644
--- a/src/asound/20-bluealsa.conf.in
+++ b/src/asound/20-bluealsa.conf.in
@@ -67,7 +67,17 @@ ctl.bluealsa {
 		}
 	}
 	type bluealsa
-	device $DEV
+	device {
+		@func refer
+		name {
+			@func concat
+			strings [
+				"bluealsa.alias."
+				$DEV
+			]
+		}
+		default $DEV
+	}
 	extended $EXT
 	bttransport $BTT
 	dynamic $DYN
@@ -135,7 +145,17 @@ pcm.bluealsa {
 	type plug
 	slave.pcm {
 		type bluealsa
-		device $DEV
+		device {
+			@func refer
+			name {
+				@func concat
+				strings [
+					"bluealsa.alias."
+					$DEV
+				]
+			}
+			default $DEV
+		}
 		profile $PROFILE
 		codec $CODEC
 		volume $VOL
```

The user then creates aliases as required, eg:

```
bluealsa.alias {
	jabra "70:BF:92:D8:82:01"
	jbl   "98:52:3D:BB:82:02"
	rpi   "B8:27:EB:B5:D7:03"
	zebra "00:1B:41:B7:31:04"
}
```

and can now use
```
aplay -D bluealsa:jbl music.wav
alsamixer -D bluealsa:DEV=jabra,EXT=battery
```

Would this be of interest to anyone else? Would it be a worth while change to BlueALSA, or maybe written up as a wiki article that users or distributions could apply themselves?

**Comment by graham8** [comment]
I can't immediately think of a use case for this for the typical user. Unless and until aplay will automatically disconnect one BT device and connect another, which will never happen.

Still, it doesn't do any harm by the looks of it.

**Comment by borine** [comment]
I've realized that the same result can be achieved by overriding parts of the bluealsa config, so there is no need to modify the original sources, therefore I am closing this discussion. For future reference, here are the ovreride definitions:
```
ctl.bluealsa.!device {
	@func refer
	name {
		@func concat
		strings [ "bluealsa.alias." $DEV ]
	}
	default $DEV
}

pcm.bluealsa.slave.pcm.!device {
	@func refer
	name {
		@func concat
		strings [ "bluealsa.alias." $DEV ]
	}
	default $DEV
}

pcm.bluealsa.slave.pcm.hint.!description {
	@func refer
	name {
		@func concat
		strings [ "bluealsa.alias." $DEV ]
	}
	default $DEV
}
```
